pkg/flowaggregator/flowaggregator.go

```diff
@@ -724,3 +895,116 @@ func (fa *flowAggregator) createInfoElementForTemplateSet(ieName string, enterpr
 	}
 	return ie, nil
 }
+
+func (fa *flowAggregator) watchConfiguration(stopCh <-chan struct{}) {
+	klog.Info("Watching for FlowAggregator configuration file")
+	for {
+		select {
+		case <-stopCh:
+			return
+		case event, ok := <-fa.configWatcher.Events:
+			if !ok {
+				klog.Error("configWatcher event channel closed")
+				return
+			}
+			// When mounted configmap is updating, watcher event "CHMOD" and "REMOVE" will happen.
+			// Need to add configFile back to watcher in order to watch the updated configMap.
+			if event.Op != fsnotify.Remove {
+				break
+			}
+			if err := fa.configWatcher.Add(fa.configFile); err != nil {
+				klog.ErrorS(err, "Error when adding configFile back to watcher")
+				return
```

Comment:
this doesn't seem to do what we discussed yesterday offline?
When such an important operation fails, we have 2 choices:
1. retry with a backoff
2. exit the process so the whole Pod can be restarted

I suggested 1 initially, but we discussed yesterday and you convinced me that 2 may be a better choice.

Right now, it seems that we exit from the function, and then we no longer try to watch the config (no retry, no restart). Am I missing something?

Reply:
Sorry. I'm still working the unit test and the latest version has not updated yet. I kill the process on my local version. I'll update it later today.

Reply:
I've updated the latest one.
Currently, I kill the fa pod for the below cases:
    1. Cannot add the configMap to watcher
    2. Cannot read the configMap
    3. Cannot Unmarshal configMap
For the validation error, I just print the error and return nil. (user can still enter the right parameters to update the fa pod)

Reply:
1 and 2 make sense to me. But I think 3 should be treated the same as a validation error. It could just be a typo in the YAML config, which makes it invalid YAML.